src/ui/component/fileListTrending/view.jsx

```diff
@@ -0,0 +1,134 @@
+// @flow
+import React, { useEffect } from 'react';
+import { FormField } from 'component/common/form';
+import FileList from 'component/fileList';
+import moment from 'moment';
+import usePersistedState from 'util/use-persisted-state';
+
+const TIME_DAY = 'day';
+const TIME_WEEK = 'week';
+const TIME_MONTH = 'month';
+const TIME_YEAR = 'year';
+const TIME_ALL = 'all';
+const TRENDING_SORT_YOU = 'you';
+const TRENDING_SORT_ALL = 'everyone';
+const TYPE_TRENDING = 'trending';
+const TYPE_TOP = 'top';
+const TYPE_NEW = 'new';
+const TRENDING_FILTER_TYPES = [TRENDING_SORT_YOU, TRENDING_SORT_ALL];
+const TRENDING_TYPES = ['trending', 'top', 'new'];
+const TRENDING_TIMES = [TIME_DAY, TIME_WEEK, TIME_MONTH, TIME_YEAR, TIME_ALL];
+
+type Props = {
+  trending: Array<string>,
+  doFetchByTags: (number, {}) => void,
+  injectedItem: any,
+  tags: Array<string>,
+  loading: boolean,
+  personal: boolean,
+};
+
+function FileListTrending(props: Props) {
+  const { doFetchByTags, trending, tags, loading, personal, injectedItem } = props;
+  const [personalSort, setPersonalSort] = usePersistedState('file-list-trending:personalSort', TRENDING_SORT_YOU);
+  const [typeSort, setTypeSort] = usePersistedState('file-list-trending:typeSort', TYPE_TRENDING);
+  const [timeSort, setTimeSort] = usePersistedState('file-list-trending:timeSort', TIME_WEEK);
+
+  const tagsString = tags.join(',');
+  useEffect(() => {
+    const options = {};
+    const newTags = tagsString.split(',');
+
+    if (personalSort === TRENDING_SORT_YOU) {
+      options.any_tags = newTags;
+    }
+
+    if (typeSort === TYPE_TRENDING) {
+      options.order_by = ['trending_global', 'trending_mixed'];
+    } else if (typeSort === TYPE_NEW) {
+      options.order_by = ['release_time'];
+    } else if (typeSort === TYPE_TOP) {
+      options.order_by = ['effective_amount'];
+      if (timeSort !== TIME_ALL) {
+        const time = Math.floor(
+          moment()
+            .subtract(1, timeSort)
+            .unix()
+        );
+        options.release_time = `>${time}`;
+      }
+    }
+
+    doFetchByTags(20, options);
+  }, [personalSort, typeSort, timeSort, doFetchByTags, tagsString]);
+
+  return (
+    <FileList
+      loading={loading}
+      uris={trending}
+      injectedItem={personalSort === TRENDING_SORT_YOU && injectedItem}
+      header={
+        <React.Fragment>
+          <h1 className={`card__title--flex`}>
+            {typeSort.charAt(0).toUpperCase() + typeSort.slice(1)} {'For'}
```

Comment:
I'd make a capitalize function since you repeat this four times. It can simply be inlined in the file if it's only used here.